kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/KafkaProtocolFilterReconciler.java

```diff
@@ -0,0 +1,210 @@
+                            filter -> {
+                                Object configTemplate = filter.getSpec().getConfigTemplate();
+                                var interpolationResult = secureConfigInterpolator.interpolate(configTemplate);
+                                return resourceNameExtractor.apply(interpolationResult)
+                                        .anyMatch(secretNameFromVolume -> secretNameFromVolume.equals(ResourcesUtil.name(secret)));
+                            });
+                    LOGGER.debug("{} {} referenced by Filters {}", secondaryClass.getName(), ResourcesUtil.name(secret), resourceIds);
+                    return resourceIds;
+                })
+                .build();
+    }
+
+    @Override
+    public UpdateControl<KafkaProtocolFilter> reconcile(
+                                                        KafkaProtocolFilter filter,
+                                                        Context<KafkaProtocolFilter> context) {
+
+        var now = ZonedDateTime.ofInstant(clock.instant(), ZoneId.of("Z"));
+
+        ConditionBuilder conditionBuilder = newResolvedRefsCondition(filter, now);
+
+        var extentSecrets = context.getSecondaryResourcesAsStream(Secret.class)
+                .map(ResourcesUtil::name)
+                .collect(Collectors.toSet());
+        LOGGER.debug("Extent secrets: {}", extentSecrets);
+
+        var extentConfigMaps = context.getSecondaryResourcesAsStream(ConfigMap.class)
+                .map(ResourcesUtil::name)
+                .collect(Collectors.toSet());
+        LOGGER.debug("Extent configmaps: {}", extentConfigMaps);
+
+        var interpolationResult = secureConfigInterpolator.interpolate(filter.getSpec().getConfigTemplate());
+        var referencedSecrets = interpolationResult.volumes().stream().flatMap(volume -> Optional.ofNullable(volume.getSecret())
+                .map(SecretVolumeSource::getSecretName)
+                .stream())
+                .collect(Collectors.toCollection(HashSet::new));
+        LOGGER.debug("Referenced secrets: {}", referencedSecrets);
+
+        var referencedConfigMaps = interpolationResult.volumes().stream().flatMap(volume -> Optional.ofNullable(volume.getConfigMap())
+                .map(ConfigMapVolumeSource::getName)
+                .stream())
+                .collect(Collectors.toCollection(HashSet::new));
+        LOGGER.debug("Referenced configmaps: {}", referencedConfigMaps);
+
+        if (extentSecrets.containsAll(referencedSecrets)
+                && extentConfigMaps.containsAll(referencedConfigMaps)) {
+            conditionBuilder.withStatus(Condition.Status.TRUE);
+        }
+        else {
+            referencedSecrets.removeAll(extentSecrets);
+            referencedConfigMaps.removeAll(extentConfigMaps);
+            String message = "Referenced";
+            if (!referencedSecrets.isEmpty()) {
+                message += " Secrets " + referencedSecrets;
```

Comment:
Let's not rely on Set's toString for our message formatting.

Reply:
FYI also switched to `TreeSet` so that the names are in a deterministic order.